### PR TITLE
REF use logging

### PR DESCRIPTION
**Pull request by beckermr**
This PR starts in on using logging in the fitting module.

### CODE REVIEW

File: ngmix/fitting.py [hunk 1]
```diff
@@ -3401,21 +3404,25 @@ def _set_minmax_err(self, min_err, max_err):
...
-        stream.write("%s\n" % None)
+        txt += "%s\n" % None
     else:
-        # fmt = ' '.join( [fmt+' ']*len(pars) )
-        # stream.write(fmt % tuple(pars))
         s = format_pars(pars, fmt=fmt)
-        stream.write(s)
-        stream.write("\n")
+        txt += s
+        txt += "\n"
```
Comment (esheldon) [inline ngmix/fitting.py:3420]: do we need the newline with the logger?

Comment (beckermr) [reply ngmix/fitting.py:3420]: ah probably not

File: ngmix/fitting.py [hunk 2]
```diff
@@ -3401,21 +3404,25 @@ def _set_minmax_err(self, min_err, max_err):
...
     """
+    txt = ""
     if front is not None:
-        stream.write(front)
-        stream.write(" ")
+        txt += front
+        txt += " "
     if pars is None:
-        stream.write("%s\n" % None)
+        txt += "%s\n" % None
```
Comment (beckermr) [inline ngmix/fitting.py:3416]:
```suggestion
        txt += "%s" % None
```

File: ngmix/fitting.py (hunk 1 above)
Comment (beckermr) [inline ngmix/fitting.py:3420]:
```suggestion
```

**Comment by esheldon** [comment]
how do we control the log level for these loggers that are created in a single module?  Do we have to set it on that logger itself or is there a way for an application to control the level of all the loggers?

**Comment by beckermr** [comment]
> how do we control the log level for these loggers that are created in a single module? Do we have to set it on that logger itself or is there a way for an application to control the level of all the loggers?

All loggers are globals. So to control the level, you get the logger for the module and set the level. 

So things like

```
import logging

logger = logging.getLogger("ngmix.fitting")
# do thing with logger
```

**Comment by esheldon** [comment]
So there is no way to turn on debug level logging everywhere?

**Comment by beckermr** [comment]
Yes. You set it for the ngmix logger from `logging.getLogger("ngmix")`

**Comment by beckermr** [comment]
Loggers also inherit settings from parents.

**Comment by esheldon** [comment]
ok, nice.

**Pull request merged**